flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/calcite/Rank.scala

```diff
@@ -34,44 +34,42 @@ import java.util
 import scala.collection.JavaConversions._
 
 /**
-  * Relational expression that returns the rows in which the rank function value of each row
+  * Relational expression that returns the rows in which the rank number of each row
   * is in the given range.
   *
-  * <p>NOTES: Different from [[org.apache.calcite.sql.fun.SqlStdOperatorTable.RANK]],
-  * [[Rank]] is a Relational expression， not a window function.
+  * The node is an optimization of `OVER` for some special cases,
+  * e.g.
+  * {{{
+  * SELECT * FROM (
+  *  SELECT a, b, RANK() OVER (PARTITION BY b ORDER BY c) rk FROM MyTable) t
+  * WHERE rk < 10
+  * }}}
+  * can be converted to this node.
   *
-  * <p>[[Rank]] will output rank function value as its last column.
-  *
-  * <p>This RelNode only handles single rank function, is an optimization for some cases. e.g.
-  * <ol>
-  * <li>
-  *   single rank function (on `OVER`) with filter in a SQL query statement
-  * </li>
-  * <li>
-  *   `ORDER BY` with `LIMIT` in a SQL query statement
-  *   (equivalent to `ROW_NUMBER` with filter and project)
-  * </li>
-  * </ol>
-  *
-  * @param cluster        cluster that this relational expression belongs to
-  * @param traitSet       the traits of this rel
-  * @param input          input relational expression
-  * @param rankFunction   rank function, including: CUME_DIST, DENSE_RANK, PERCENT_RANK, RANK,
-  *                       ROW_NUMBER
-  * @param partitionKey   partition keys (may be empty)
-  * @param sortCollation  order keys for rank function
-  * @param rankRange      the expected range of rank function value
+  * @param cluster          cluster that this relational expression belongs to
+  * @param traitSet         the traits of this rel
+  * @param input            input relational expression
+  * @param partitionKey     partition keys (may be empty)
```

Comment:
Can this be empty?

Reply:
in order to convert sql like: 
`select * from (
  select a, b, rank() over (order by b) as rk from MyTable
) where rk < a` to `Rank` on Stream, partition keys may be empty .

or we do not convert this kind of sql to `Rank`, keep it as `Over`